apps/storybook/src/Selection.stories.tsx

```diff
@@ -101,11 +151,95 @@ ChangeSelectionColors.args = {
   color: 'blue',
 };
 ChangeSelectionColors.argTypes = {
+  selection: SelectionArgType,
   color: {
     control: { type: 'color' },
   },
 };
 
+export const SelectToZoom: Story<{
+  modifierKey?: 'Alt' | 'Control' | 'Shift';
+}> = (args) => {
+  const { modifierKey } = args;
+  return (
+    <VisCanvas
+      abscissaConfig={{ visDomain: [-5, 5], showGrid: true }}
+      ordinateConfig={{ visDomain: [-0.5, 1.5], showGrid: true }}
+    >
+      <PanMesh />
+      <ZoomMesh />
+      <ZoomSelectionMesh modifierKey={modifierKey} />
+      <GaussianCurve />
+    </VisCanvas>
+  );
+};
+SelectToZoom.args = {
+  modifierKey: 'Control',
+};
+SelectToZoom.argTypes = {
+  modifierKey: modifierKeyArgType,
+};
+
+function DataSelectionComponent(props: {
+  selection: 'line' | 'rectangle';
+  startPoint: Vector2;
+  endPoint: Vector2;
+}) {
+  const { selection, startPoint, endPoint } = props;
+  const { dataToWorld } = useAxisSystemContext();
+  const SelectionComponent =
+    selection === 'line' ? SelectionLine : SelectionRect;
+
+  return (
+    <SelectionComponent
+      startPoint={dataToWorld(startPoint)}
+      endPoint={dataToWorld(endPoint)}
+    />
+  );
+}
+
+export const PersistSelection: Story<TemplateProps> = (args) => {
+  const { selection, modifierKey } = args;
+  const [start, setStart] = useState<Vector2>();
+  const [end, setEnd] = useState<Vector2>();
+
+  const SelectionComponent =
+    selection === 'line' ? SelectionLine : SelectionRect;
+
+  return (
+    <VisCanvas
+      abscissaConfig={{ visDomain: [-5, 5], showGrid: true }}
+      ordinateConfig={{ visDomain: [-0.5, 1.5], showGrid: true }}
+    >
+      {/* <PanMesh />
+      <ZoomMesh /> */}
```

Comment:
In case this flew under your radar, it might be worth uncommenting these two meshes.